internal/handlers/utils_test.go

```diff
@@ -0,0 +1,137 @@
+	registryURL string
+	ctx         context.Context
+}
+
+func setupPrivateRegistry(t *testing.T) *RegistryTestSuite {
+	registryContainer, err := registry.Run(
+		t.Context(),
+		registry.DefaultImage,
+		registry.WithHtpasswd("user:$2y$10$nTQigvLRGGHCBQwZB4MPPe2SA6GYG218uTe1ntHusNcEjLaAfBive"), // user:password
+		testcontainers.WithWaitStrategy(
+			wait.ForHTTP("/v2/").
+				WithPort(registryPort).
+				WithStatusCodeMatcher(func(status int) bool {
+					// Registry should return 401 (Unauthorized) for unauthenticated requests
+					return status == http.StatusUnauthorized
+				}).
+				WithStartupTimeout(5*time.Minute),
+		),
+	)
+	require.NoError(t, err)
+
+	// Get registry URL
+	registryHostAddress, err := registryContainer.HostAddress(t.Context())
+	require.NoError(t, err)
+
+	cleanup, err := registry.SetDockerAuthConfig(
+		registryHostAddress, authUser, authPass,
+	)
+	require.NoError(t, err)
+	defer cleanup()
+
+	imageRefPull := fmt.Sprintf("%s/%s:%s", registryURI, imageName, tag)
+	t.Logf("Pulling image from external registry: %s", imageRefPull)
+	err = pullImage(t.Context(), imageRefPull)
+	require.NoError(t, err)
+
+	imageRef := fmt.Sprintf("%s/%s:%s", registryContainer.RegistryName, imageName, tag)
+	t.Logf("Tagging image %s to %s", imageRefPull, imageRef)
+	err = tagImage(t.Context(), imageRefPull, imageRef)
+	require.NoError(t, err)
+
+	t.Logf("Pushing image to registry: %s", imageRef)
+	err = registryContainer.PushImage(t.Context(), imageRef)
+	require.NoError(t, err)
+
+	return &RegistryTestSuite{
+		registry:    registryContainer,
+		registryURL: registryHostAddress,
+		ctx:         t.Context(),
+	}
+}
+
+// TODO(alegrey91): fix upstream
+// pullImage pulls an image from an external Registry.
+// It will use the internal registry to store the image.
+func pullImage(ctx context.Context, ref string) error {
+	dockerProvider, err := testcontainers.NewDockerProvider()
+	if err != nil {
+		return fmt.Errorf("failed to create Docker provider: %w", err)
+	}
+	defer dockerProvider.Close()
+
+	dockerCli := dockerProvider.Client()
+
+	pullOpts := image.PullOptions{
+		All: true,
+	}
+
+	pullOutput, err := dockerCli.ImagePull(ctx, ref, pullOpts)
+	if err != nil {
+		return fmt.Errorf("failed to push image %s: %w", ref, err)
+	}
+
+	_, err = io.Copy(io.Discard, pullOutput)
+	if err != nil {
+		return fmt.Errorf("failed to read output: %w", err)
+	}
+
+	return nil
+}
+
+// TODO(alegrey91): fix upstream
+// tagImage tags an image from the local Registry.
+func tagImage(ctx context.Context, image, ref string) error {
+	dockerProvider, err := testcontainers.NewDockerProvider()
+	if err != nil {
+		return fmt.Errorf("failed to create Docker provider: %w", err)
+	}
+	defer dockerProvider.Close()
+
+	dockerCli := dockerProvider.Client()
+
+	err = dockerCli.ImageTag(ctx, image, ref)
+	if err != nil {
+		return fmt.Errorf("failed to tag image %s: %w", image, err)
+	}
+
+	return nil
+}
+
+func (suite *RegistryTestSuite) tearDown(t *testing.T) {
```

Comment:
let's call this `stop`. it should accept ctx.Context and return error so we can remove testing.T